Zendesk (Source) initial commit

mohan-h opened the pull request:
Zendesk (Source) initial commit

carlojavier commented on help/sources/tutorials/ui/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,141 @@
+description: Learn how to create a Zendesk source connection using the Adobe Experience Platform UI.
+hide: true
+hidefromtoc: true
+exl-id: 
```
We can remove "exl-id" here as it is auto-generated upon publishing

carlojavier commented on help/sources/tutorials/ui/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,141 @@
+
+[Zendesk](https://www.zendesk.com) is a popular customer service solution and sales tool.
+
+This [!DNL Adobe Experience Platform] [!DNL sources connector](https://experienceleague.adobe.com/docs/experience-platform/sources/home.html?lang=en) leverages the [!DNL Zendesk Search API > Export Search Results](https://developer.zendesk.com/api-reference/ticketing/ticket-management/search/#export-search-results) that returns users information into Experience Platform from Zendesk for further p ...
```
We can remove the [!DNL] tags here for Adobe Experience Platform and source connector. Also let's reword "sources" to just sources. The [!DNL] tag in "Zendesk Search API" can also be removed.

carlojavier commented on help/sources/tutorials/ui/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,141 @@
+The following items are required before you start configuring the extension:
+
+* You need to have a Zendesk Support account. 
+    * Go to the Zendesk [register](https://www.zendesk.com/register/) page to register and create a Zendesk account, if you do not have one already.
```
Since there is only one prerequisite, we can restructure this to just one short paragraph. Something like: "Before you can configure a Zendesk source connection, you must first ensure that you have Zendesk Support account. If you do not have a Zendesk Support account, go to the [Zendesk register](https://www.zendesk.com/register/) to register and create your Zendesk Support account before attempting to do this tutorial.

carlojavier commented on help/sources/tutorials/ui/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,141 @@
+| subdomain | Unique domain specific to your account created during the registration process. | *xxxxx.zendesk.com*|
+| API token | Zendesk API token.<br></br>Navigate to the Zendesk website and access **[!UICONTROL Settings]** > **[!UICONTROL Apps and Integrations]** > **[!UICONTROL Zendesk API]** page. Then Navigate to the **[!UICONTROL API Tokens]** section. |*0lZnClEvkJSTQ7olGLl7PMhVq99gu26GTbJtf*|
+
+![Zendesk domain](../../../../images/tutorials/create/zendesk/zendesk-domain.png?lang=en)
```
We can probably just skip this domain screenshot and just add a link to "Zendesk website" in the credential tables

carlojavier commented on help/sources/tutorials/ui/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,141 @@
+
+1. In the Platform UI, select **[!UICONTROL Datasets]** from the left navigation bar to access the [!UICONTROL Datasets] workspace. The [!UICONTROL Dataset Activity] screen displays the details of executions.
+![Activity page](../../../../images/tutorials/create/zendesk/dataset-activity.png?lang=en)
+![Dataflow page](../../../../images/tutorials/create/zendesk/dataflow-monitoring.png?lang=en)
```
Let's add a sentence between these two screenshots, describing the step that transitions between them. Something along the lines of: "Next, select the dataflow run ID of the dataflow that you want to view, to see specific details about that dataflow run."

carlojavier commented on help/sources/tutorials/ui/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,141 @@
+![Activity page](../../../../images/tutorials/create/zendesk/dataset-activity.png?lang=en)
+![Dataflow page](../../../../images/tutorials/create/zendesk/dataflow-monitoring.png?lang=en)
+
+1. Next, select the **[!UICONTROL Data governance]** tab, you will see the schema details.
```
We can rephrase this to something like: "the **[!UICONTROL Data governance]** tab to see the schema details."

carlojavier commented on help/sources/tutorials/ui/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,141 @@
+
+### Zendesk schema
+
+* A list of supported mappings that are to be set up for the [!DNL Zendesk Search API > Export Search Results](https://developer.zendesk.com/api-reference/ticketing/ticket-management/search/#export-search-results) is given below.
```
Can we rephrase this to something like: "The table below outlines a list of supported mappings that must be set up for the [!DNL Zendesk Search API > Export Search Results](https://developer.zendesk.com/api-reference/ticketing/ticket-management/search/#export-search-results)."

carlojavier commented on help/sources/tutorials/api/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,1159 @@
+description: Learn how to connect Adobe Experience Platform to Zendesk using the Flow Service API.
+hide: true
+hidefromtoc: true
+exl-id: 
```
We can remove "exl-id" here as it is auto-generated.

carlojavier commented on help/sources/tutorials/api/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,1159 @@
+
+[Zendesk](https://www.zendesk.com) is a popular customer service solution and sales tool.
+
+This [!DNL Adobe Experience Platform] [!DNL sources connector](https://experienceleague.adobe.com/docs/experience-platform/sources/home.html?lang=en) leverages the [!DNL Zendesk Search API > Export Search Results](https://developer.zendesk.com/api-reference/ticketing/ticket-management/search/#export-search-results) that returns users information into Experience Platform from Zendesk for further p ...
```
We can remove the three DNL tags in this paragraph and let's reword "sources" to just "source"

carlojavier commented on help/sources/tutorials/api/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,1159 @@
+| subdomain | Unique domain specific to your account created during the registration process. | *xxxxx.zendesk.com*|
+| API token | Zendesk API token.<br></br>Navigate to the Zendesk website and access **[!UICONTROL Settings]** > **[!UICONTROL Apps and Integrations]** > **[!UICONTROL Zendesk API]** page. Then Navigate to the **[!UICONTROL API Tokens]** section. |*0lZnClEvkJSTQ7olGLl7PMhVq99gu26GTbJtf*|
+
+![Zendesk domain](../../../../images/tutorials/create/zendesk/zendesk-domain.png?lang=en)
```
Like the UI doc, we can remove this domain screenshot and just add a link to "Zendesk website" in the screenshot

carlojavier reviewed (changes requested):
Hey @mohan-h thanks for this! Added some comments, just some small changes. Thanks!

mohan-h committed:
Merge branch 'AdobeDocs:main' into zendesk-source

mohan-h committed:
Zendesk schema table modified and list of customers added

Zendesk schema table modified to not display the target mapping.
Image indicating customer list added.

mohan-h committed:
Image updates after publishing.

Image updates after publishing.

vivetiwa commented on help/sources/tutorials/ui/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,134 @@
+
+| Credential | Description | Example |
+| --- | --- | --- |
+| subdomain | Unique domain specific to your account created during the registration process. <br/><br/> Refer to the [Zendesk documentation](https://support.zendesk.com/hc/en-us/articles/4409381383578-Where-can-I-find-my-Zendesk-subdomain-) if you require any guidance. | *xxxxx.zendesk.com*|
```
It should be inform of url, https should be added to host with the subdomain.
https://support.zendesk.com/hc/en-us/articles/4409381383578-Where-can-I-find-my-Zendesk-subdomain-#:~:text=https%3A//yoursubdomain.zendesk.com

vivetiwa commented on help/sources/tutorials/ui/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,134 @@
+
+## Next steps
+
+By following this tutorial, you have established a connection to your *Zendesk* account. You can now continue on to the next tutorial and [configure a dataflow to bring data into Platform](https://experienceleague.adobe.com/docs/experience-platform/sources/ui-tutorials/dataflow/crm.html).
```
@carlojavier for rest based connector preview screen input will be somewhat different. There would be no table or folder selection as it is in CRM and cloud based sources.

In rest user either needs to provide some input parameters or if input params are not there  then he will see the preview of the data.

Should be there preview screen in documentation as we have added for mailchimp campaign similar to that section defining the preview of zendesk ? - 

https://experienceleague.adobe.com/docs/experience-platform/sources/ui-tutorials/create/marketing-automation/mailchimp-campaigns.html?lang=en#:~:text=Once%20your%20source%20is%20authenticated

carlojavier replied:
@vivetiwa good call, thanks for heads up on this. We'll definitely update this page to be similar to the mailchimp doc structure

mohan-h replied:
@carlojavier sorry didn't understand need guidance.

vivetiwa commented on help/sources/tutorials/api/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,1152 @@
+    "mappings": [
+        {
+            "sourceType": "ATTRIBUTE",
+            "source": "results.id",
```
`results` should be replace with `result` as it would be single object not an array.

carlojavier replied:
Does this apply to all values for `source` in this section or just `results.id`?

mohan-h replied:
@vivetiwa This was the exact response from the curl request

vivetiwa commented on help/sources/tutorials/api/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,1152 @@
+        "name": "Zendesk base connection",
+        "description": "Zendesk base connection to authenticate to Platform",
+        "connectionSpec": {
+            "id": "2526ee57-0af4-46ea-a477-720a53108145",
```
connectionSpec id should be - `0a27232b-2c6e-4396-b8c6-c9fc24e37ba4`

vivetiwa commented on help/sources/tutorials/api/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,1152 @@
+            "version": "1.0"
+        },
+        "auth": {
+            "specName": "OAuth generic-rest-connector",
```
specName should be - `OAuth2 Refresh Code`

vivetiwa commented on help/sources/tutorials/api/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,1152 @@
+        "auth": {
+            "specName": "OAuth generic-rest-connector",
+            "params": {
+                "host":{ZENDESK_SUBDOMAIN}
```
missing double quotes - `"host": "{ZENDESK_SUBDOMAIN}"`

mohan-h replied:
Good spot.

vivetiwa commented on help/sources/tutorials/api/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,1152 @@
+| `description` | An optional value that you can include to provide more information on your base connection. |
+| `connectionSpec.id` | The connection specification ID of your source. This ID can be retrieved after your source is registered and approved through the [!DNL Flow Service] API. |
+| `auth.specName` | The authentication type that you are using to authenticate your source to Platform. |
+| `auth.params.` | Contains the credentials required to authenticate your source. |
```
`auth.params.` params missing.
 Please refer : https://experienceleague.adobe.com/docs/experience-platform/sources/api-tutorials/create/marketing-automation/mailchimp-campaign.html?lang=en#:~:text=authenticate%20your%20source%20to%20Platform.-,auth.params.host,-The%20root%20URL%20used%20to

mohan-h replied:
Thanks @vivetiwa, added.
@carlojavier the template doesn't provide guidance on this. Suggest to add guidance.

vivetiwa commented on help/sources/tutorials/api/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,1152 @@
+    "schema": {
+        "type": "object",
+        "properties": {
+            "results": {
```
`results` should be `result`

mohan-h replied:
curl response has the attribute as results.

vivetiwa commented on help/sources/tutorials/api/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,1152 @@
+                "has_more": true,
+                "after_cursor": "eyJmaWVsZCI6ImNyZWF0ZWRfYXQiLCJkZXNjIjp0cnVlLCJ0aWVCcmVha0ZpZWxkIjoiaWQiLCJ0aWVCcmVha0Rlc2MiOmZhbHNlLCJzb3J0VmFsdWVzIjpbMTY1MjE3MjA4NTAwMCw2MDMwODg3MTM5NzI5XSwiZXhwb3J0ZWRUaHVzRmFyIjoxMDAsInNlc3Npb25TdGFydCI6MTY1MjQzNDE4MzkwOCwiY3JlYXRlZEF0IjoxNjUyNDM0MTgzOTI2LCJzYWx0ZWRSZXF1ZXN0SGFzaCI6LTczMzQxODQ0NSwic2FsdGVkQ3Vyc29ySGFzaCI6MjgxNzM0NzU2fQ=="
+            },
+            "links": {
```
links, meta should not be visible as part of preview. Could be removed from preview.

vivetiwa commented on help/sources/tutorials/api/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,1152 @@
+            "links": {
+                "next": "https://d3v-dxinfosys.zendesk.com/api/v2/search/export.json?filter%5Btype%5D=user&page%5Bafter%5D=eyJmaWVsZCI6ImNyZWF0ZWRfYXQiLCJkZXNjIjp0cnVlLCJ0aWVCcmVha0ZpZWxkIjoiaWQiLCJ0aWVCcmVha0Rlc2MiOmZhbHNlLCJzb3J0VmFsdWVzIjpbMTY1MjE3MjA4NTAwMCw2MDMwODg3MTM5NzI5XSwiZXhwb3J0ZWRUaHVzRmFyIjoxMDAsInNlc3Npb25TdGFydCI6MTY1MjQzNDE4MzkwOCwiY3JlYXRlZEF0IjoxNjUyNDM0MTgzOTI2LCJzYWx0ZWRSZXF1 ...
+            },
+            "results": {
```
`results` to `result`

mohan-h replied:
Checked the curl response, it is `results`. I'll share the API request, response to you on slack.

vivetiwa commented on help/sources/tutorials/api/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,1152 @@
+        "description": "Zendesk Source Connection",
+        "baseConnectionId": "70383d02-2777-4be7-a309-9dd6eea1b46d",
+        "connectionSpec": {
+            "id": "6360f136-5980-4111-8bdf-15d29eab3b5a",
```
connectionSpec id should be - `0a27232b-2c6e-4396-b8c6-c9fc24e37ba4`

vivetiwa commented on help/sources/tutorials/api/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,1152 @@
+        "xdmSchema": "https://ns.adobe.com/{TENANT_ID}/schemas/995dabbea86d58e346ff91bd8aa741a9f36f29b1019138d4",
+        "xdmVersion": "1.0",
+        "id": null,
+        "mappings": [
```
duplicate "mappings" attributes

mohan-h replied:
Good spot.

vivetiwa commented on help/sources/tutorials/api/create/customer-success/zendesk.md:
```diff
@@ -0,0 +1,1152 @@
+        "version": 0,
+        "xdmSchema": "https://ns.adobe.com/{TENANT_ID}/schemas/995dabbea86d58e346ff91bd8aa741a9f36f29b1019138d4",
+        "xdmVersion": "1.0",
+        "id": null,
```
`"id": null` could be removed

mohan-h committed:
Updates from review comments

Updates from review comments

mohan-h committed:
Fixing a developer reference

Fixing a developer reference

mohan-h committed:
Merge branch 'main' into zendesk-source

mohan-h committed:
Merge branch 'AdobeDocs:main' into zendesk-source

mohan-h committed:
Merge branch 'AdobeDocs:main' into zendesk-source

mohan-h committed:
Merge branch 'AdobeDocs:main' into zendesk-source

mohan-h committed:
Cleanup w.r.t images

1. authentication-existing-account.png has been superseded by existing.png
2. authentication-new-account.png has been superseded by new.png
3. masked identifiable information in new.png
4. Removed the reference to the data governance bullet

juliette-massardier committed:
Update license-usage.md

Hello, as discussed with the Adobe Journey Optimizer product team, we should add the AJO licenses listed here and the link to the AJO product description. Thanks! Juliette (AJO doc team)

JCON3DEV committed:
Merge pull request #80 from juliette-massardier/patch-1

Update AJO license-usage.md

mohan-h committed:
Merge branch 'AdobeDocs:main' into zendesk-source

mohan-h commented:
The documentation for the connector has been made live at:
https://experienceleague.adobe.com/docs/experience-platform/sources/connectors/customer-success/zendesk.html?lang=en
